src/components/BottomBar/WWTPanel.jsx

```diff
@@ -0,0 +1,332 @@
+          targetData: target,
+          selectedTarget: camera.selectedTargetId,
+          selectedBrowser: camera.selectedBrowserId,
+          isUsingRae: camera.isUsingRadiusAzimuthElevation,
+          isFacingCamera: camera.isFacingCamera,
+          cameraInSolarSystem: camera.cameraInSolarSystem,
+          moduleIsLoaded: true
+        });
+      }
+    } catch (e) {
+      console.log(e);
+      // Stop the timer to get the target data
+      clearInterval(this.targetDataID);
+      this.setState({
+        moduleIsLoaded: false
+      });
+    }
+  }
+
+  getSelectedTargetImages() {
+    const { systemList } = this.props;
+    const { targetData, selectedBrowser } = this.state;
+    const selectedImagesIndices = targetData[selectedBrowser];
+
+    if (!selectedImagesIndices) {
+      return [];
+    }
+
+    // For some reason, ghoul sends this vector as an object. Convert to array
+    const images = selectedImagesIndices.selectedImages;
+    if (!images) return [];
+    const indices = Object.values(images);
+    return indices.map(index => systemList[index.toString()]);
+  }
+
+  getAllImages() {
+    const images = this.props.systemList;
+    return images || {};
+  }
+
+  getCurrentTargetColor() {
+    const { targetData, selectedBrowser } = this.state;
+    const browser = targetData[selectedBrowser];
+
+    return browser ? `rgb(${browser.color})` : 'gray';
+  }
+
+  getNearestImages() {
+    const { systemList } = this.props;
+    const { targetData, selectedBrowser } = this.state;
+    const targetPoint = targetData[selectedBrowser];
+    if (!targetPoint) {
+      return [];
+    }
+
+    const searchRadius = targetPoint.FOV / 2;
+
+    const isWithinFOV = (coord, target, FOV) => (coord < (target + FOV) && coord > (target - FOV));
+
+    // Only load images that have coordinates within current window
+    const imgsWithinTarget = systemList.filter((img) => {
+      if (!img.hasCelestialCoords) {
+        return false; // skip
+      }
+      if (isWithinFOV(img.ra, targetPoint.ra, searchRadius)
+          && isWithinFOV(img.dec, targetPoint.dec, searchRadius)) {
+        return true;
+      }
+      return false;
+    });
+
+    const distPow2 = (a, b) => (a - b) * (a - b);
+
+    const euclidianDistance = (a, b) => {
+      let sum = 0;
+      for (let i = 0; i < 3; i++) {
+        sum += distPow2(a.cartesianDirection[i], b.cartesianDirection[i]);
+      }
+      return Math.sqrt(sum);
+    };
+
+    imgsWithinTarget.sort((a, b) => {
+      const result = euclidianDistance(a, targetPoint) > euclidianDistance(b, targetPoint);
+      return result ? 1 : -1;
+    });
+    return imgsWithinTarget;
+  }
+
+  // Setters
+  setCurrentTabHeight(height) {
+    this.setState({ currentTabHeight: height });
+  }
+
+  setCurrentPopoverHeight(height) {
+    this.setState({ currentPopoverHeight: height });
+  }
+
+  selectImage(identifier) {
+    if (identifier) {
+      this.setState({
+        imageName: identifier,
```

Comment:
This state variable should probably be called something like `activeImageNave`, to clarify that it's the active/selected image